### ISSUE TITLE
Variable.Key() returning non zero value when missing in material.json

### ISSUE THREAD

**Issue by marandra** [KratosMultiphysics/Kratos#1587]
I understand that CHECK_VARIABLE_KEY function should check the presence of a parameter checking that its .Key() value is not zero. In this case, the Key() value is not zero when missing in the material.json file.

Tried within StructuralMechanicsApplication. This is the Check function of the CL (KRATOS_WATCH added for this example):
```
int LinearJ2Plasticity3D::Check(const Properties& rMaterialProperties,
                                                 const GeometryType& rElementGeometry,
                                                 const ProcessInfo& rCurrentProcessInfo)
{
    KRATOS_CHECK_VARIABLE_KEY(YOUNG_MODULUS);
    KRATOS_WATCH(YOUNG_MODULUS.Key()); //for debug only
    KRATOS_CHECK(0); //for debug only
    
    KRATOS_CHECK_VARIABLE_KEY(POISSON_RATIO);
    KRATOS_CHECK_VARIABLE_KEY(YIELD_STRESS);
    KRATOS_CHECK_VARIABLE_KEY(REFERENCE_HARDENING_MODULUS);
    KRATOS_CHECK_VARIABLE_KEY(ISOTROPIC_HARDENING_MODULUS);
    KRATOS_CHECK_VARIABLE_KEY(INFINITY_HARDENING_MODULUS);
    KRATOS_CHECK_VARIABLE_KEY(HARDENING_EXPONENT);

    return 0;
}
```
These are the last messages. I understand that YOUNG_MODULUS.Key() should be 0:
```
YOUNG_MODULUS.Key() : 1169
Traceback (most recent call last):
  File "MainKratos.py", line 77, in <module>
    solver.Initialize()
  File "/home/mraschi/Projects/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_static_solver.py", line 63, in Initialize
    super(StaticMechanicalSolver, self).Initialize() # The mechanical solver is created here.
  File "/home/mraschi/Projects/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py", line 213, in Initialize
    self.Check()
  File "/home/mraschi/Projects/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py", line 254, in Check
    self.get_mechanical_solver().Check()
RuntimeError: Error: Check failed because 0 is not true
```

This is the material.json file, where YOUNG_MODULUS is missing:
```
{  
    "properties": [{
        "model_part_name": "MATRIX",
        "properties_id": 1,
        "Material": {
            "name": "matrix",
            "constitutive_law": {
                "name": "KratosMultiphysics.StructuralMechanicsApplication.LinearJ2Plasticity3DLaw"
            },
            "Variables": {
                "POISSON_RATIO": 0.3,
                "YIELD_STRESS": 5.5,
                "REFERENCE_HARDENING_MODULUS": 1.0,
                "ISOTROPIC_HARDENING_MODULUS": 0.12924,
                "INFINITY_HARDENING_MODULUS": 0.0,
                "HARDENING_EXPONENT": 1.0
                },
            "Tables":  {}
            }
    }, {
        "model_part_name": "INCLUSION",
        "properties_id": 2,
        "Material": {
            "name": "inclusion",
            "constitutive_law": {
                "name": "KratosMultiphysics.StructuralMechanicsApplication.LinearJ2Plasticity3DLaw"
            },
            "Variables": {
                "POISSON_RATIO": 0.2,
                "YIELD_STRESS": 2.5,
                "REFERENCE_HARDENING_MODULUS": 1.0,
                "ISOTROPIC_HARDENING_MODULUS": 0.12924,
                "INFINITY_HARDENING_MODULUS": 0.0,
                "HARDENING_EXPONENT": 1.0
                },
            "Tables":  {}
            }
    }]
}

```

**Comment by jcotela** [comment]
'YOUNG_MODULUS.Key()' returns the internal key for the variable 'YOUNG_MODULUS', which is a property of the variable (and its registration into kratos) and unrelated to json input. The only "realistic" situation in which this key could be zero is, for application variables, the case where you didn't import the application that defines the variable before its use in the code.

Note that this shoud become obsolete once the new variable registration is merged into core (in  the new system, the key will be determined at compile time, and therefore can never be zero).

**Comment by marandra** [comment]
Ahhh, I see. Thanks @jcotela for the clarification.
